src/GraphQlSdk.php

```diff
@@ -328,6 +333,41 @@ public function createPayment($paymentData)
         return $this->doRequest($query, $paymentData);
     }
 
+    /**
+     * @param string $paymentId
+     * @param string $orderId
+     * @param string $authorizationResponse
+     * @param string|null $threeDSecureResponse
+     * @return array|false
+     * @throws \Exception
+     */
+    public function confirmCardAuthorization(
+        string $paymentId,
+        string $orderId,
+        string $authorizationResponse,
+        ?string $threeDSecureResponse
+    ) {
+        $query = <<<'QUERY'
+        mutation cardAuthorizationConfirm ($input: CardAuthorizationConfirmInput!) {
+            cardAuthorizationConfirm (input: $input) {
+                authorizationId
+            }
+        }
+QUERY;
+        $variables = [
+            "input" => [
+                "merchantId" => $this->merchantId,
+                "orderId" => $orderId,
+                "paymentId" => $paymentId,
+                "authorizationResponse" => $authorizationResponse,
+                "threeDSecureResponse" => $threeDSecureResponse,
+            ],
+        ];
+
+        return $this->doRequest($query, $variables)["data"]["cardAuthorizationConfirm"];
```

Comment:
One of the error flows relies on checking the error code string inside the graphql response.
In the other plugin we had to modify the doRequest method to return an api exception which had both the error message and code. 
Then the plugin is able to use that in the error handling. 

If you look for this in the other codebase you'll see how its done there
```
$errorString = $processed["errors"][0]["message"];
$errorCode = $processed["errors"][0]["extensions"]["errorCode"];
throw new ApiError($errorCode, $errorString);
```